include/cuda/stream_view

```diff
@@ -0,0 +1,97 @@
+//===----------------------------------------------------------------------===//
+//
+// Part of libcu++, the C++ Standard Library for your entire system,
+// under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef _CUDA_STREAM
+#define _CUDA_STREAM
+
+#include "std/cstddef"
+#include "std/detail/__config"
+#include "std/detail/__pragma_push"
+#include "std/version"
+#include "detail/__cuda_util"
+
+#include <cuda_runtime_api.h>
+#include <stdexcept>
+
+_LIBCUDACXX_BEGIN_NAMESPACE_CUDA
+
+/**
+ * @brief A non-owning wrapper for a `cudaStream_t`.
+ *
+ */
+class stream_view {
+public:
```

Comment:
We should have some type alias here that provides the type that `get` will return, to allow folks to write generic code on this category of things (they could just `decltype`, but a type alias is still nice). I imagine that we'll have a very similar thing for `event` which will also have a `get` method.

Possibly names would be:
* `[native_]handle_type`
* `value_type`
* `pointer`

Reply:
Added `stream_view::value_type`.